sql/src/main/java/io/crate/analyze/GeneratedColumnComparisonReplacer.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.ImmutableMap;
+import io.crate.analyze.relations.AnalyzedRelation;
+import io.crate.analyze.relations.GeneratedColumns;
+import io.crate.analyze.symbol.*;
+import io.crate.metadata.*;
+import io.crate.operation.operator.AndOperator;
+import io.crate.operation.operator.EqOperator;
+import io.crate.operation.operator.Operators;
+import io.crate.types.DataTypes;
+import org.elasticsearch.common.inject.Singleton;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+
+@Singleton
+public class GeneratedColumnComparisonReplacer {
+
+    private static final Predicate<Symbol> IS_GENERATED_COLUMN = new Predicate<Symbol>() {
+        @Override
+        public boolean apply(@Nullable Symbol input) {
+            assert input != null;
```

Comment:
no need for the assertion,  ``input instanceof Reference`` will return false if input is null